### PR TITLE
Remove the extra slash while loading resources.

### PR DESCRIPTION
**Pull request by line-bear**
## Description
The new default config file already contains / at the end of path string, which may cause double slash while loading resources, It seems to work fine at the moment, but it is recommended to remove it to prevent unpredictable situations.

## Type of changes

- [x] Bug fix
- [ ] New feature 
- [ ] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

**Comment by ghost** [comment]
We can not be sure that someone editing config file will remember to add `/` at the end of string.

File: src/main/java/emu/grasscutter/Configuration.java [hunk 1]
```diff
@@ -48,15 +48,15 @@ public final class Configuration extends ConfigContainer {
      */
     
     public static String DATA(String path) {
-        return DATA_FOLDER + "/" + path;
+        return DATA_FOLDER + path;
```
Comment (ghost) [inline src/main/java/emu/grasscutter/Configuration.java]:
```suggestion
        return Paths.get(DATA_FOLDER, path).toString();
```

File: src/main/java/emu/grasscutter/Configuration.java [hunk 2]
```diff
@@ -48,15 +48,15 @@ public final class Configuration extends ConfigContainer {
...
     }
     
     public static String RESOURCE(String path) {
-        return RESOURCES_FOLDER + "/" + path;
+        return RESOURCES_FOLDER + path;
```
Comment (ghost) [inline src/main/java/emu/grasscutter/Configuration.java]:
```suggestion
        return Paths.get(RESOURCES_FOLDER, path).toString();
```

File: src/main/java/emu/grasscutter/Configuration.java [hunk 3]
```diff
@@ -48,15 +48,15 @@ public final class Configuration extends ConfigContainer {
...
     }
     
     public static String SCRIPT(String path) {
-        return SCRIPTS_FOLDER + "/" + path;
+        return SCRIPTS_FOLDER + path;
```
Comment (ghost) [inline src/main/java/emu/grasscutter/Configuration.java]:
```suggestion
        return Paths.get(SCRIPTS_FOLDER, path).toString();
```

**Review by ghost — CHANGES_REQUESTED** [review]
Let's use proper path joining.

```java
import java.nio.file.Paths;
```

**Comment by ghost** [comment]
You should also perform same changes in all other files where `*_FOLDER` are used. Most likely we should mark them as private and instead provide helper functions like `DATA` for all other ones that don't have one currently.

**Review by line-bear — COMMENTED** [review]
Use path join to prevent double slash

**Comment by line-bear** [comment]
> You should also perform same changes in all other files where `*_FOLDER` are used. Most likely we should mark them as private and instead provide helper functions like `DATA` for all other ones that don't have one currently.

all done, have fun

**Review by line-bear — COMMENTED** [review]
Mark fields as private to prevent use *FOLDER directly

File: src/main/java/emu/grasscutter/utils/Crypto.java [hunk 4]
```diff
@@ -20,11 +20,11 @@ public final class Crypto {
 	public static byte[] ENCRYPT_SEED_BUFFER = new byte[0];
 	
 	public static void loadKeys() {
-		DISPATCH_KEY = FileUtils.read(KEYS_FOLDER + "/dispatchKey.bin");
-		DISPATCH_SEED = FileUtils.read(KEYS_FOLDER + "/dispatchSeed.bin");
+		DISPATCH_KEY = FileUtils.read(KEY("/dispatchKey.bin"));
```
Comment (ghost) [inline src/main/java/emu/grasscutter/utils/Crypto.java]: No need for slash here

**Comment by ghost** [comment]
Thanks. Looks good to me now.

I'll most likely go around whole codebase in another PR later and see if we can refactor `Utils.toFilePath` to be a wrapper for path join instead of dumb replacing `/` with separator. I've also spotted few places that forget to use `Utils.toFilePath` so will need to handle that too.

**Comment by line-bear** [comment]
> Thanks. Looks good to me now.
> 
> I'll most likely go around whole codebase in another PR later and see if we can refactor `Utils.toFilePath` to be a wrapper for path join instead of dumb replacing `/` with separator. I've also spotted few places that forget to use `Utils.toFilePath` so will need to handle that too.

Waitting for your message :)

**Review by ghost — APPROVED** [review]

**Pull request merged**